docs/sources/tutorials/create-alerts-with-logs/index.md

```diff
@@ -0,0 +1,394 @@
+```
+
+
+
+## Optional: Use promtail with a sample log-generating script
+
+This optional step uses a python script to generate the sample logs used in this tutorial to create alerts.
+
+1. [Install promtail](https://grafana.com/docs/loki/latest/clients/promtail/installation/) on your local machine and configure it to send logs to your Loki instance.
+1. Install Python3 on your local machine if needed.
+1. Copy the python script below and paste it into a new file on your local machine.
+
+    ```#!/bin/env python3
+    ​
+    import datetime
+    import math
+    import random
+    import sys
+    import time
+
+
+    ​# Simulation parameters
+    requests_per_second = 2
+    failure_rate = 0.05
+    get_post_ratio = 0.9
+    get_average_duration_ms = 500
+    post_average_duration_ms = 2000
+    ​
+    ​
+    while True:
+    # Exponential distribution random value of average 1/lines_per_second.
+    d = random.expovariate(requests_per_second)
+    time.sleep(d)
+    if random.random() < failure_rate:
+        status = "500"
+    else:
+        status = "200"
+    if random.random() < get_post_ratio:
+        method = "GET"
+        duration_ms = math.floor(random.expovariate(1/get_average_duration_ms))
+    else:
+        method = "POST"
+        duration_ms = math.floor(random.expovariate(1/post_average_duration_ms))
+    timestamp = datetime.datetime.now(tz=datetime.timezone.utc).isoformat()
+    print(f"{timestamp} level=info method={method} url=/ status={status} duration={duration_ms}ms")
+    sys.stdout.flush()
+    ```
+
+1. Give the script executable permissions.
+
+    In a terminal window on linux-based systems run the command:
+
+    ```
+    chmod 755  ./web-server-logs-simulator.py
+    ```
+
+1. Run the script.
+
+    * Use `tee` to direct the script output to the console and the specified file path. For example, if promtail is
+    configured to monitor `/var/log` for `.log` files you can direct the script output to `/var/log/web_requests.log` file.
+
+    * To avoid running the script with elevated permissions, create the log file manually and change the permissions for the output file only.
+
+        ```
+        sudo touch /var/log/web_requests.log
+        chmod 755 /var/log/web_requests.log
+        python3 ./web-server-logs-simulator.py | tee -a /var/log/web_requests.log
+        ```
+
+1. Verify that the logs are showing up in Grafana’s Explore view:
+
+    * Navigate to explore in Grafana.
+    * Select the Loki datasource from the drop-down.
+    * Check the toggle for **builder | code** in the top right corner of the query box and switch the query mode to builder if it’s not already selected.
+    * Select the filename label from the drop-down and choose your `web_requests.log` file from the value drop-down.
+    * Click **Run Query**.
+    * You should see logs and a graph of log volume.
+
+#### Troubleshooting the script
```

Comment:
I think you want this to be either an H3 or H2, depending on whether it should be nested under `## Optional: Use promtail with a sample log-generating script` or on the same level, respectfully